src/backend/app/s3.py

```diff
@@ -112,8 +112,8 @@ def get_obj_from_bucket(bucket_name: str, s3_path: str) -> BytesIO:
     try:
         response = client.get_object(bucket_name, s3_path)
         return BytesIO(response.read())
-    except Exception as e:
-        raise ValueError(str(e)) from e
+    except Exception:
+        return None
```

Comment:
I think the exception should remain here, as other things may rely on the information if download fails:
```python
log.warning(f"Failed attempted download from S3 path: {s3_path}")
raise ValueError(str(e)) from e
```

Then you an put the exception catch higher up in the submission logic:
```python
try:
    data = await get_obj_from_bucket(xxx, xxx)
except ValueError as e:
    log.debug(f"Submission data does not exist in S3 for project {project_id}")
    data = None
```

Perhaps we even want to halt execution and raise an error to the user instead (depending on the context)?

```python
try:
    data = await get_obj_from_bucket(xxx, xxx)
except ValueError as e:
    msg = f"Submission data does not exist in S3 for project {project_id}"
    log.debug(msg)
    raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail=msg) from e
```

Reply:
usually we dont have submissions in s3, so most of the time there wont be any objects related to that project. In that case we need to update submissions in s3. if we raise error from this point , it wont update submissions in s3 at very first place.

Reply:
Ah that makes sense. In this case, the error handling would be:

```python
try:
    data = await get_obj_from_bucket(xxx, xxx)
except ValueError as e:
    msg = f"Submission data does not exist in S3 for project {project_id}. Continuing to creation"
    log.debug(msg)
```

Reply:
I found the main issue is elsewhere, i think some changes needs to be done in osm_fieldwork, i will let you know after i completely track the issue and debug it.

Reply:
the issue was in downloadThread ,
previously ,
```python
subs = form.listSubmissions(project_id, task, filters)
if type(subs) == dict:
    log.error(f"{subs['message']}, {subs['code']} ")
    continue
# log.debug(f"There are {len(subs)} submissions for {task}")
if len(subs) > 0:
    data += subs
```
here subs is always dict so it obstructs data to be appended.
i am thinking of solution like this which solves this issue.
```python
subs = form.listSubmissions(project_id, task, filters)
if subs == {}:
    log.error(f"{subs['message']}, {subs['code']} ")
    continue
# log.debug(f"There are {len(subs)} submissions for {task}")
if len(subs["value"]) > 0:
    data += subs["value"]
```
Since we want to append only the values of key "value" of submissions and upload to s3.

Reply:
> Ah that makes sense. In this case, the error handling would be something like:
> 
> ```python
> try:
>     data = await get_obj_from_bucket(xxx, xxx)
> except ValueError as e:
>     log.debug(f"Submission data does not exist in S3 for project ({project_id}). Data will be generated.")
> ```

it will make error message static , i dont think we should do this, since other errors may also be raised from `get_obj_from_bucket(xxx, xxx)`. i think we should leave it as it is.